.github/workflows/main.yml

```diff
@@ -12,37 +12,52 @@ jobs:
         python-version: [3.7]
 
     steps:
-    - uses: actions/checkout@v1
-    - name: Set up Python ${{ matrix.python-version }}
-      uses: actions/setup-python@v1
-      with:
-        python-version: ${{ matrix.python-version }}
-
-    - name: Test formatting
-      run: |
-        pip install black
-        black --check .
-
-    - name: Install dependencies
-      run: |
-        python -m pip install --upgrade pip
-        pip install -r requirements.txt
-
-    - name: Test with pytest
-      run: |
-        pip install pytest
-        pytest
-
-    - name: Publish to Docker Registry
-      uses: elgohr/Publish-Docker-Github-Action@master
-      with:
-        name: simplelogin/app-ci
-        username: ${{ secrets.DOCKER_USERNAME }}
-        password: ${{ secrets.DOCKER_PASSWORD }}
-
-    - name: Send Telegram message
-      uses: appleboy/telegram-action@master
-      with:
-        to: ${{ secrets.TELEGRAM_TO }}
-        token: ${{ secrets.TELEGRAM_TOKEN }}
-        args: Docker image pushed on ${{ github.ref }}
+      - uses: actions/checkout@v2
+        
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+
+
+      - uses: actions/cache@v1
+        with:
+          path: ~/.cache/pip
+          key: ${{ runner.os }}-pip-${{ hashFiles('**/requirements.txt') }}
+          restore-keys: |
+            ${{ runner.os }}-pip-
+
+      - name: Test formatting
+        run: |
+          pip install black
+          black --check .
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r requirements.txt
+
+      - name: Test with pytest
+        run: |
+          pip install pytest
+          pytest
+
+      - name: Extract tag name
+        shell: bash
+        run: echo ::set-output name=tag::$(echo ${GITHUB_REF##*/})
```

Comment:
It is here: https://stackoverflow.com/a/58034787/13459000

You will probably notice that there is a comment discussed an issue of this with `Gitflow-style`, however, I intentionally ignored that issue since Docker doesn't allow `/` appears in the tag name (so if we fix that probably it will fail the CI).